File: tests/testdata/codegen/aws-resource-options-4.26-pp/nodejs/aws-resource-options-4.26.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";
import * as aws from "@khulnasoft/aws";

const provider = new aws.Provider("provider", {region: "us-west-2"});
const bucket1 = new aws.s3.Bucket("bucket1", {}, {
    provider: provider,
    dependsOn: [provider],
    protect: true,
    ignoreChanges: [
        "bucket",
        "lifecycleRules[0]",
    ],
});

File: tests/testdata/codegen/aws-resource-options-4.26-pp/go/aws-resource-options-4.26.go
package main

import (
	"github.com/khulnasoft/khulnasoft-aws/sdk/v4/go/aws"
	"github.com/khulnasoft/khulnasoft-aws/sdk/v4/go/aws/s3"
	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

func main() {
	khulnasoft.Run(func(ctx *khulnasoft.Context) error {
		provider, err := aws.NewProvider(ctx, "provider", &aws.ProviderArgs{
			Region: khulnasoft.String("us-west-2"),
		})
		if err != nil {
			return err
		}
		_, err = s3.NewBucket(ctx, "bucket1", nil, khulnasoft.Provider(provider), khulnasoft.DependsOn([]khulnasoft.Resource{
			provider,
		}), khulnasoft.Protect(true), khulnasoft.IgnoreChanges([]string{
			"bucket",
			"lifecycleRules[0]",
		}))
		if err != nil {
			return err
		}
		return nil
	})
}
